components/brave_today/browser/brave_news_controller.cc

```diff
@@ -630,6 +630,14 @@ void BraveNewsController::ConditionallyStartOrStopTimer() {
       timer_prefetch_.Start(FROM_HERE, base::Minutes(1), this,
                             &BraveNewsController::Prefetch);
     }
+
+    GetPublishers(base::BindOnce([](BraveNewsController* controller, Publishers publishers) {
```

Comment:
This is the actual fix: Push publishers to all listeners on opt-in

Reply:
I'm curious what would happen if there's no in-memory publisher data and it takes too long for the api request helper in publisher controller to get the publisher data. Shortly, would the NTP crash if the callback here is invoked later than the page loading?

Reply:
So the actual crash was being caused because the page was asking for some suggestions, and getting them back. However, we hadn't pushed the publishers to the front end at that point, so we had an empty list of publishers, but a bunch of suggestedPublisherIds.

`GetSuggestedPublisherIds` depends on `GetPublishers`, so if we get suggestions then the api request for publishers **must** have completed.

So to answer your question, no :smile:

Reply:
Thanks @fallaciousreasoning ! It can't happen logically.